app/javascript/components/subject/Jurisdiction.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Modal, Form } from 'react-bootstrap';
+import _ from 'lodash';
+import axios from 'axios';
+
+import InfoTooltip from '../util/InfoTooltip';
+import reportError from '../util/ReportError';
+
+class Jurisdiction extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showJurisdictionModal: false,
+      jurisdiction_path: this.props.jurisdictionPaths[this.props.patient.jurisdiction_id],
+      original_jurisdiction_id: this.props.patient.jurisdiction_id,
+      validJurisdiction: true,
+      apply_to_group: false,
+      loading: false,
+      reasoning: '',
+    };
+    this.origState = Object.assign({}, this.state);
```

Comment:
Maybe it's just cause Im more familiar with Lodash, but I always prefer using `_.cloneDeep( value )` instead of `Object.assign` to prevent any possible data-losses with cloning. Not a huge thing (especially in this file), but it's an option.

Reply:
We use `Object.assign` like this in other files too so I'm going to leave this as is for now